File: scripts/generate_jwt.py
#!/usr/bin/env python3
"""
Generate JWT tokens for load testing
Usage: python generate_jwt.py <userId>
"""

import sys
import os
from datetime import datetime, timedelta
import jwt

def generate_token(user_id: str) -> str:
    """Generate JWT token valid for 24 hours"""
    secret = os.environ.get('JWT_SECRET')
    if not secret:
        raise ValueError("JWT_SECRET environment variable not set")
    
    now = datetime.utcnow()
    exp = now + timedelta(hours=24)
    
    payload = {
        'sub': user_id,
        'iat': now,
        'exp': exp
    }
    
    token = jwt.encode(payload, secret, algorithm='HS256')
    return token

if __name__ == '__main__':
    if len(sys.argv) < 2:
        print("Usage: python generate_jwt.py <userId>", file=sys.stderr)
        sys.exit(1)
    
    user_id = sys.argv[1]
    token = generate_token(user_id)
    print(token)

File: scripts/GenerateJwt.java
import io.jsonwebtoken.Jwts;
import io.jsonwebtoken.SignatureAlgorithm;
import io.jsonwebtoken.security.Keys;
import java.nio.charset.StandardCharsets;
import java.util.Date;
import javax.crypto.SecretKey;

public class GenerateJwt {
    public static void main(String[] args) {
        if (args.length < 1) {
            System.err.println("Usage: java GenerateJwt <userId>");
            System.exit(1);
        }
        
        String userId = args[0];
        String secret = System.getenv("JWT_SECRET");
        if (secret == null || secret.isBlank()) {
            System.err.println("JWT_SECRET environment variable not set");
            System.exit(1);
        }
        
        SecretKey key = Keys.hmacShaKeyFor(secret.getBytes(StandardCharsets.UTF_8));
        
        // Token valid for 24 hours
        long now = System.currentTimeMillis();
        long exp = now + (24 * 60 * 60 * 1000);
        
        String token = Jwts.builder()
            .setSubject(userId)
            .setIssuedAt(new Date(now))
            .setExpiration(new Date(exp))
            .signWith(key, SignatureAlgorithm.HS256)
            .compact();
        
        System.out.println(token);
    }
}
